### PR TITLE
Image probe output enhancements

### PR DESCRIPTION
**Pull request by gurevichmark**
Suppress some image probe output messages to be only in verbose mode and skip node objects of type switch

### CODE REVIEW

**Comment by whowutwut** [comment]
When this is ready for review/merge, can you add in some sample output so I can easily see the externals

**Comment by gurevichmark** [comment]
Second pull request to display a summary output if number of nodes > 20 and to display a list of nodes for a specific UUID.

Detailed output (when number of nodes <= 20):
```
[root@fs2vm100 subcmds]# xcatprobe image -c
---- Gathering information from all diskless pingable compute nodes ----
Not all diskless pingable compute nodes are installed with the same os image.                                     [FAIL]
         RHEL72diskless2vm1 rhels7.2-ppc64le-netboot-compute : b1948b38-486e-4fb9-9c49-1add5959fe2e
         RHEL72diskless2vm2 rhels7.2-ppc64le-netboot-compute : b1948b38-486e-4fb9-9c49-1add5959fe2e
         RHEL72disklessvm1 rhels7.2-ppc64le-netboot-compute : 5f8aac4c-16d3-42ee-b05c-014099a89a4e
         RHEL72disklessvm2 rhels7.2-ppc64le-netboot-compute : 5f8aac4c-16d3-42ee-b05c-014099a89a4e
         RHEL72disklessvm3 rhels7.2-ppc64le-netboot-compute : 5f8aac4c-16d3-42ee-b05c-014099a89a4e
         RHEL72disklessvm4 rhels7.2-ppc64le-netboot-compute : 5f8aac4c-16d3-42ee-b05c-014099a89a4e

```
Summary output format (when number of nodes > 20):
```
[root@fs2vm100 subcmds]# xcatprobe image -c
---- Gathering information from all diskless pingable compute nodes ----
Not all diskless pingable compute nodes are installed with the same os image.                                     [FAIL]
         4 compute nodes running OS=>rhels7.2-ppc64le-netboot-compute UUID=>5f8aac4c-16d3-42ee-b05c-014099a89a4e
         2 compute nodes running OS=>rhels7.2-ppc64le-netboot-compute UUID=>b1948b38-486e-4fb9-9c49-1add5959fe2e
Run 'xcatprobe image -u UUID' to display a list of compute nodes that have installed OS with specified UUID

```

Output listing nodes for a specified UUID:
```
[root@fs2vm100 subcmds]# xcatprobe image -u 5f8aac4c-16d3-42ee-b05c-014099a89a4e
---- Gathering information from all diskless pingable compute nodes ----
Compute nodes running OS=>rhels7.2-ppc64le-netboot-compute UUID=>5f8aac4c-16d3-42ee-b05c-014099a89a4e:
RHEL72disklessvm1
RHEL72disklessvm2
RHEL72disklessvm3
RHEL72disklessvm4

```

File: xCAT-probe/subcmds/image [hunk 1]
```diff
@@ -336,11 +353,45 @@ if ($CONSISTENCY_CHECK) {
...
+                #Produce summary output for a large system
+                foreach $os_uuid (sort keys %unique_image_hash) {
+                    my ($os_name, $uuid) = split(":", $os_uuid);
+                    my $count = scalar(split(",",$unique_image_hash{$os_uuid}));
+                    $node_image_table .= sprintf("         $count compute nodes running OS=>$os_name UUID=>$uuid\n");
+                }
+                $node_image_table .= "Run 'xcatprobe image -u UUID' to display a list of compute nodes that have installed OS with specified UUID";
+            }
+            else {
+                #Produce list output for small system
```
Comment (whowutwut) [inline xCAT-probe/subcmds/image]: same format as indicated above

**Comment by whowutwut** [comment]
First off, this is great!  :+1:  

Some general comments, not all need to be resolved in this pull request... 
* I think we are using the term "compute" but really this can be used over any noderange, so the node doesn't necessarily need to be a compute node.  Thinking maybe we should drop that word.
* "diskless pingable" is not flowing well :-) 
   Perhaps we can mention that at the beginning, then just drop it from the rest,.   So: 
 `---- Gathering information from all diskless nodes (pingable) ----`  Then simply write `Not all diskless nodes are installed with the same os image.`
*

**Comment by gurevichmark** [comment]
@whowutwut Suggested changes implemented under a new commit.

**Comment by whowutwut** [comment]
Thanks, let's merge this so we can give it a try

**Comment by whowutwut** [comment]
Fixes issues #1940

**Pull request merged**